docs/content/docs/build/auth.md

```diff
@@ -101,13 +101,50 @@ provider is useful for testing, but in production scenarios you must use a secur
 
 ### A secure serverless token provider
 
-One option for building a secure token provider is to create a serverless Azure Function and expose it as a token
-provider. This enables you to store the *tenant secret key* on a secure server. Your application calls the Azure Function to
-generate tokens rather than signing them locally like the `InsecureTokenProvider` does.
+One option for building a secure token provider is to create a serverless Azure Function and expose it. This enables you to store the *tenant secret key* on a secure server. Your application calls the Azure Function to generate tokens rather than signing them locally like the `InsecureTokenProvider` does.
+
+You can create the token provider class say `AzureFunctionTokenProvider` which extends the `ITokenProvider` interface without exposing the tenant key secret in client-side code. This class would be responsible for fetching the token from your very own backend. It accepts the url to Azure function and an optional user object.
+
+```typescript
+import { ITokenProvider, ITokenResponse } from "@fluidframework/azure-client";
+
+export class AzureFunctionTokenProvider implements ITokenProvider {
+  constructor(
+    private readonly azFunctionUrl: string,
+    private readonly user?: Pick<AzureMember, "userId" | "userName" | "additionalDetails">,
+  );
+
+  fetchOrdererToken(tenantId: string, documentId: string, refresh?: boolean): Promise<ITokenResponse>;
+
+  fetchStorageToken(tenantId: string, documentId: string, refresh?: boolean): Promise<ITokenResponse>;
```

Comment:
Do these intentionally omit the implementation? If so, I suggest making the code reflect that, and add a comment like `// omitted` so it's clear.